### PR TITLE
Add Dynatrace sampler to OpenTelemetry Tracing Provider

### LINKED ISSUE #50001: OpenTelemetryTracingProvider: Allow configuring samplers
**Describe the feature request**

Recently we contributed new features on Istio to improve the support for OpenTelemetry tracing in Istio. [OTLP/HTTP Exporter](https://github.com/istio/istio/pull/49052) and [Resource Detectors](https://github.com/istio/istio/pull/49204).

We want to add the last piece of our contributions from Envoy to istio: The ability to configure samplers to the OpenTelemetry tracing provider. 

The OpenTelemetry project defines a [Sampler API](https://opentelemetry.io/docs/specs/otel/trace/sdk/#sampler) and this API is available [now in Envoy](https://www.envoyproxy.io/docs/envoy/v1.29.2/api-v3/config/trace/opentelemetry/samplers). With the next Envoy release, it will also come with a [Dynatrace Sampler](https://github.com/envoyproxy/envoy/pull/32598). To make it clear, the API can be implemented by anyone, so it is expected that more samplers are added in Envoy. 

[I had a discussion](https://github.com/istio/api/pull/3002#discussion_r1445079905) with @howardjohn in one of the PRs on how the sampler would interact with the existing ways of configuring sampling in Istio, and the conclusion we arrived was:

```
provider.sampler > telemetry.RandomSamplingPercentage > defaultConfig.tracing.sampling > PILOT_TRACE_SAMPLING
```

Which means:

- If `OpenTelemetryTracingProvider` has a custom sampler configured: 
  - Set `telemetry.RandomSamplingPercentage` to `100` as all spans should arrive on the sampler, so it can perform its algorithm. This is hardcoded and the user cannot change (doesn't make sense to). Without all spans a sampler can't make consistent sampling decisions. 

- Else continue with the same precedence as today, essentially nothing changes when users don't use a Sampler: 
```
 telemetry.RandomSamplingPercentage > defaultConfig.tracing.sampling > PILOT_TRACE_SAMPLING
```

**Affected product area (please put an X in all that apply)**

[ ] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

CC @zirain @howardjohn

### PR DESCRIPTION
**Pull request by joaopgrassi**
**Please provide a description of this PR:**

Closes #50001 

This is a follow up on the recent API change to allow configuring custom Samplers to the OpenTelemetry Tracing Provider in Istio. https://github.com/istio/api/pull/3134

This PR then uses this new API config and allow configuring the first custom sampler, the Dynatrace sampler. A bit of context: Today, there's multiple ways to configure sampling in Istio, which works/have the following precedence:

```
telemetry.RandomSamplingPercentage > defaultConfig.tracing.sampling > PILOT_TRACE_SAMPLING
```

This PR adds another option, changing the order above to: 

```
provider.sampler > telemetry.RandomSamplingPercentage > defaultConfig.tracing.sampling > PILOT_TRACE_SAMPLING
```

Which means:

- If `OpenTelemetryTracingProvider` has a custom sampler configured (**what this PR adds**): 
  - Set `telemetry.RandomSamplingPercentage` to `100` as all spans should arrive on the sampler, so it can perform its algorithm. This is hardcoded and the user cannot change (doesn't make sense to). Without all spans a sampler can't make consistent sampling decisions. 

- Else, continue with the same precedence as today. Essentially nothing changes when users don't use a Sampler: 

```
 telemetry.RandomSamplingPercentage > defaultConfig.tracing.sampling > PILOT_TRACE_SAMPLING
```

cc @howardjohn @zirain 

I plan to add a new documentation page to explain how the ways sampling can be configured when using the OpenTelemetry tracer in Istio.

### CODE REVIEW

**Comment by joaopgrassi** [comment]
Hi @howardjohn @zirain this is the implementation of samplers we have been talking. It would be great if we can get this in time for the 1.22 release. Thank you!

File: releasenotes/notes/dynatrace-sampler.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: telemetry
+issue:
+- 50001
+releaseNotes:
+  - |
+    **Enabled** configuring Dynatrace Sampler for the `OpenTelemetryTracingProvider` in `MeshConfig`.
```
Comment (zirain) [inline releasenotes/notes/dynatrace-sampler.yaml:8]: Thanks!

**Pull request merged**